src/core/service.c

```diff
@@ -1561,21 +1561,47 @@ static int service_spawn(
                 }
         }
 
-        if (flags & EXEC_SETENV_RESULT) {
-                if (asprintf(our_env + n_env++, "SERVICE_RESULT=%s", service_result_to_string(s->result)) < 0)
-                        return -ENOMEM;
+        if ((flags & EXEC_SETENV_RESULT) ^ (flags & EXEC_SETENV_MONITOR_RESULT)) {
```

Comment:
I'm not too sure of this style, we don't usually do it - the usual style would be adding an assert that both flags cannot be true at the same time, and then just a normal `||` here

Reply:
I've got rid of this, back to normal bitwise OR and an assert.